src/func/Microsoft.Health.Fhir.Ingest.Host/Startup.cs

```diff
@@ -0,0 +1,37 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Ingest.Config;
+using Microsoft.Health.Fhir.Ingest.Telemetry;
+
+[assembly: FunctionsStartup(typeof(Microsoft.Health.Fhir.Ingest.Service.Startup))]
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            EnsureArg.IsNotNull(builder, nameof(builder));
+
+            builder.Services.AddSingleton<ITelemetryLogger>(sp =>
+            {
+                var telemetryConfiguration = new TelemetryConfiguration();
+                telemetryConfiguration.TelemetryInitializers.Add(new OperationCorrelationTelemetryInitializer());
+                telemetryConfiguration.InstrumentationKey = Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY") ?? string.Empty;
```

Comment:
Should we make it fail on missing that configuration?

Reply:
Yes, that makes sense. I was thinking we did not want it to fail when people are running the function locally and potentially not providing an app insights instrumentation key. But I'll update this to specifically check AZURE_FUNCTIONS_ENVIRONMENT = development, to not have the function fail when running locally.

Reply:
Updated to only configure app insights when the setting is not null, but there were some challenges with logging errors during startup. In particular there wouldn't be a place to easily view the error/exception. I have chosen to let the functions startup when the setting isn't populated, because that seemed marginally better than throwing an exception and having no place to view the exception